cloud/services/scalesets/vmss_test.go

```diff
@@ -717,6 +717,188 @@ func TestReconcileVMSS(t *testing.T) {
 				s.SetProvisioningState(infrav1.VMStateSucceeded)
 			},
 		},
+		{
+			name:          "can create a vmss with user assigned identity",
```

Comment:
*No action in the PR, but more for maintainers.*

Dang... these tests are verbose with only like 2 lines that need to be updated. It would be nice to dry these up.

cc: @CecileRobertMichon and @nader-ziada